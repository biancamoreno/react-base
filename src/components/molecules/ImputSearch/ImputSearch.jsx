import React, { Component } from "react";
import PropTypes from "prop-types";

import "./ImputSearch.scss";

class ImputSearch extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        const { type, placeholder, handleChange, handleSearch } = this.props;

        return (
            <div className="imput-search__wrapper">
                <input
                    type = {type}
                    placeholder = {placeholder}
                    className = "imput-search"
                    onChange = {handleChange}
                    />
                <button className="imput-search__button" onClick = {handleSearch}>
                    <img className="imput-search__icon" src="../src/assets/images/ic_Search.png" />
                </button>
            </div>
        )
    }
}

ImputSearch.propTypes = {
    placeholder: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    handleChange: PropTypes.func.isRequired,
    handleSearch: PropTypes.func.isRequired
}

export default ImputSearch;
