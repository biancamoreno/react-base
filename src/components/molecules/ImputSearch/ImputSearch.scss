@import '../../../assets/variables/variables.scss';

.imput-search__wrapper { 
    display: flex;
    flex-wrap: nowrap; 
}

.imput-search {
    width: 100%;
    height: 32px;
    font-size: 18px;
    padding: 0 10px;
    border: none;
    border-top-left-radius: 3px;
    border-bottom-left-radius: 3px;
}

.imput-search__button {
    padding: 0 10px;
    border: none;
    background-color: $light-gray;
    border-top-right-radius: 3px;
    border-bottom-right-radius: 3px;
}