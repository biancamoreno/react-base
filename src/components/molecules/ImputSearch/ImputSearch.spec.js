import React from 'react';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme';
import ImputSearch from './ImputSearch';

describe('MyComponent', () => {
  it('should render correctly with props', () => {
    const props = {
      type: '',
      placeholder: '',
      handleChange: jest.fn(),
      handleSearch: jest.fn()
    }

    const wrapper = shallow(<ImputSearch {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  // it('should render correctly with no props', () => {
  //   const component = shallow(<ImputSearch/>);
  //   expect(component).toMatchSnapshot();
  // });
});