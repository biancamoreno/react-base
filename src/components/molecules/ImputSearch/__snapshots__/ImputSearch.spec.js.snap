// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly with props 1`] = `
<div
  className="imput-search__wrapper"
>
  <input
    className="imput-search"
    onChange={[MockFunction]}
    placeholder=""
    type=""
  />
  <button
    className="imput-search__button"
    onClick={[MockFunction]}
  >
    <img
      className="imput-search__icon"
      src="../src/assets/images/ic_Search.png"
    />
  </button>
</div>
`;
